crates/sparrow-scheduler/src/queue.rs

```diff
@@ -0,0 +1,148 @@
+//! Provide the global and per-thread (local) work queues.
```

Comment:
These are relatively simple wrappers around the existing `work_queue` crate. It provides:

1. A fixed set of interfaces / documentation
2. Some tests around ordering

The idea is that by *not* working with the crate directly we should be more easily able to use other queues if we discover the need.